bokeh/models/annotations.py

```diff
@@ -1096,6 +1097,42 @@ class Whisker(Annotation):
 
     level = Override(default="underlay")
 
+class InfoPane(Annotation):
+    ''' Renders a info-pane.
+
+    .. note::
+        This model is currently managed by BokehJS and is not useful
+        directly from python.
+
+    '''
+    level = Override(default="overlay")
+
+    anchor = Enum(TooltipAttachment, help="""
+    Whether the pane should be anchored at the left or right of the pane
+    or above or below it, or if it should be automatically placed
+    in the horizontal or vertical dimension.
+    """)
+
+    x_anchor = Float(default=0, help="""
+    Initial anchor value for pane to apply to the x-coordinate.
+    """)
```

Comment:
So these are in screen units? Let's wait to see what @bryevdv thinks, but I think you should be able to specify these in data units as well as screen units:

```python
anchor_units = Enum(SpatialUnit, ...
```

You'll then have to implement that transform client side.

Reply:
That sounds good we can provide the users to be able to specify data units as well. Though, a bit unsure about how the user would benefit from it as our objective is to place it arbitrarily anywhere unless we use it for some tool I guess.

Reply:
By default a user will want to place the tooltip in data units so unless I'm mistaken that the current default is to place in screen units is actually less useful for users.

Reply:
Oh, I get it. I shall update to look for providing data units as well.

Reply:
I think we need to support both. Users will typically want to annotate things like a scatter data point or a point on a line, for which they will only have data space coordinates. But there may be times when users want to position an info pane at a specific fixed location e.g. "20 pixels off the top left corner". Tools will also most naturally want to be able to address to info pane using screen coordinates.